test/y2network/sysconfig/connection_config_writer_test.rb

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "../../test_helper"
+
+require "y2network/sysconfig/connection_config_writer"
+require "y2network/sysconfig/connection_config_writers/eth"
+require "y2network/connection_config/ethernet"
+require "y2network/interface_type"
+
+describe Y2Network::Sysconfig::ConnectionConfigWriter do
+  subject(:writer) { described_class.new }
+
+  describe "#write" do
+    let(:handler) do
+      instance_double(
+        Y2Network::Sysconfig::ConnectionConfigWriters::Eth,
+        write: nil
+      )
+    end
+
+    let(:conn) do
+      instance_double(
+        Y2Network::ConnectionConfig::Ethernet,
+        interface: "eth0",
+        type:      Y2Network::InterfaceType::ETHERNET
+      )
+    end
+
+    let(:file) do
+      instance_double(
+        Y2Network::Sysconfig::InterfaceFile, save: nil, clean: nil
+      )
+    end
```

Comment:
I would really feel more confident if we can simply switch SCR here and operate on real files. So we see what is real changes.

Reply:
We rely on `Y2Network::Sysconfig::InterfaceFile` in which unit tests we switch SCR and operate on real files. See https://github.com/yast/yast-network/blob/fc128f537beb6f84a72783c08f4a121a13616ee8/test/y2network/sysconfig/interface_file_test.rb#L37.